rust/kernel/src/chrdev.rs

```diff
@@ -38,7 +38,7 @@ impl Builder {
         self
     }
 
-    pub fn build(self) -> KernelResult<Registration> {
+    pub fn build(self, this_module: *mut bindings::module) -> KernelResult<Registration> {
```

Comment:
We need a safer wrapper around this, otherwise passing the raw ptr makes this interface unsafe.

Reply:
Passing the address is not unsafe on its own. I agree that it could look better (e.g. a `struct ThisModule` as I mentioned in the message above).

Reply:
Because this will deref the ptr, and _creating_ a raw ptr is safe, the impact is that this is unsound :-)

Reply:
Yeah, it is definitely unsound, but the interface isn't unsafe due to the pointer passing on its own -- i.e. if we could check the pointer (somehow), then the function wouldn't need to be `unsafe` to be sound.

Related: at some point we will have to establish some guidelines in the docs (before Rust code starts growing quickly with people calling kernel APIs directly etc.) on "how much soundness-correct" we want to be (see similar discussions in e.g. cxx).

Reply:
Sure, but there's no way to check if the pointer is correct :-)

In general, safe interfaces can't really take raw pointers for this reason.

Reply:
I know, that is why I said "somehow" :-)

I think in my first reply you understood I was disagreeing about the function needing to be marked unsafe or having a proper abstraction -- I was only (pedantically) pointing out that it is not the passing but what we do with the pointer that makes the function unsafe.